internal/controller/storageclient_controller.go

```diff
@@ -235,6 +235,17 @@ func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
 					return reconcile.Result{}, fmt.Errorf("failed to reconcile cephConnection: %v", err)
 				}
 			}
+		case "Subscription":
+			var subscriptionSpec *opv1a1.SubscriptionSpec
+			if err := json.Unmarshal(eResource.Data, &subscriptionSpec); err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to unmarshall subscriptionSpec error: %v", err)
+			}
+
+			if utils.AddAnnotation(r.storageClient, utils.DesiredSubscriptionChannelAnnotationKey, subscriptionSpec.Channel) {
+				if err := r.update(r.storageClient); err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to update StorageClient with subscription channel annotation: %v", err)
+				}
+			}
```

Comment:
This entire change does not make sense. What we want ti achieve is the provider sending the desired subscription and the client code takes that and update the exiting subscription

Reply:
@nb-ohad The existing code is simply adding the annotation on the storage client resource. So I have done the same here. I have checked with @leelavg regarding why we are not setting the channel in the subscription itself he explained to me with a scenario
-> Assume we have a client connected to 2 different providers and one of the providers is upgraded and the other provider is not in that case it's not good to upgrade the client and the status reporter is also not a good place to do all this logic hence it is used to add just annotation.  this logic is part of operatorconfig_controller [here](https://github.com/red-hat-storage/ocs-client-operator/blob/main/internal/controller/operatorconfigmap_controller.go#L218)

Reply:
But the subscription you are sending is the OCS subscription, not the client op subscription. I would expect the client-op subscription to be sent and blindly merged with the existing subscription.

Reply:
Over huddle, its been decided to patch ocs-client-subs blindly and from the provider send the spec of ocs-client-operator.

Reply:
If there is a mismatch among other storageclients then the patch will not go through webhook and the reconciler will not make progress until the patch is successful.

If we don't want to consider more than one storageclient then the whole webhook mechanism doesn't need to be there in the first place.